tools/arcilator/arcilator.cpp

```diff
@@ -375,6 +414,36 @@ static LogicalResult processBuffer(
   if (failed(pmLlvm.run(module.get())))
     return failure();
 
+#ifdef ARCILATOR_ENABLE_JIT
+  // Handle JIT execution.
+  if (!runSimulation.empty()) {
+    mlir::ExecutionEngineOptions engineOptions;
+    engineOptions.jitCodeGenOptLevel = llvm::CodeGenOptLevel::Aggressive;
+
+    auto executionEngine =
+        mlir::ExecutionEngine::create(module.get(), engineOptions);
+    if (!executionEngine) {
+      llvm::Error err = executionEngine.takeError();
+      llvm::errs() << "failed to create execution engine: " << err << "\n";
+      llvm::consumeError(std::move(err));
+      return failure();
+    }
+
+    auto expectedFunc = (*executionEngine)->lookupPacked(runSimulation);
+    if (!expectedFunc) {
+      llvm::Error err = expectedFunc.takeError();
+      llvm::errs() << "failed to run simulation: " << err << "\n";
+      llvm::consumeError(std::move(err));
+      return failure();
+    }
+
+    void (*simulationFunc)(void **) = *expectedFunc;
+    (*simulationFunc)(nullptr);
+
+    return success();
```

Comment:
That’s a great question. My original approach was to have the JIT be another flavor of the emit flag, but then it was a bit weird to specify an entry point.

But perhaps I could do that and default the entry point to `main`, with the ability to use a second flag to set the entry point! What do you think?

Reply:
Sounds good to me. Maybe the default should be `entry` (or something else) though. Because with the JIT you can also emit the object file which could be an option to skip the separate opt and llc calls. But the `main` function needs to have two arguments which the entry points must not. We could then just insert the main function as a wrapper around `entry`